File: app/src/main/java/nl/rsdt/japp/jotial/maps/searching/Searchable.java
package nl.rsdt.japp.jotial.maps.searching;



import java.util.List;

import nl.rsdt.japp.jotial.maps.wrapper.IMarker;

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 29-7-2016
 * Description...
 */
public interface Searchable {

    List<String> provide();

    IMarker searchFor(String query);

}

File: app/src/main/java/nl/rsdt/japp/jotial/maps/searching/Searchable.kt
package nl.rsdt.japp.jotial.maps.searching


import nl.rsdt.japp.jotial.maps.wrapper.IMarker

/**
 * @author Dingenis Sieger Sinke
 * @version 1.0
 * @since 29-7-2016
 * Description...
 */
interface Searchable {

    fun provide(): List<String>

    fun searchFor(query: String): IMarker

}
